obstore/python/obstore/auth/earthdata.py

```diff
@@ -6,157 +6,434 @@
 from __future__ import annotations
 
 import json
+import os
 from datetime import datetime
 from typing import TYPE_CHECKING
+from urllib.parse import urlparse
+
+from obstore.auth._http import default_aiohttp_session, default_requests_session
 
 if TYPE_CHECKING:
+    import sys
+    from collections.abc import Mapping
+
+    if sys.version_info >= (3, 10):
+        from typing import Never
+    else:
+        from typing_extensions import Never
+
+    import aiohttp
+    import aiohttp_retry
+    import requests
+
     from obstore.store import S3Config, S3Credential
 
-CREDENTIALS_API = "https://archive.podaac.earthdata.nasa.gov/s3credentials"
+
+EARTHDATA_HOST_OPS = "urs.earthdata.nasa.gov"
+"""Hostname of the NASA Earthdata Login Operational (OPS) system.
+
+This is the default host used during credential authorization.
+"""
+
+EARTHDATA_HOST_UAT = "uat.urs.earthdata.nasa.gov"
+"""Hostname of the NASA Earthdata Login User-Acceptance Testing (UAT) system."""
```

Comment:
Can you say when a user might want to use this instead?

Is this only for testing? If it's not intended for end-user use, we should remove it from here.

Reply:
This is not often used, but it is something that may still want to expose to users. Users of the `earthaccess` library have expressed the desire to hit "test" data rather than "production" data, and both hosts are coded into that library.

However, I'm not opposed to removing this. Users who want to hit the UAT system will know the host name and can simply provide it explicitly themselves, so I'd be okay with having only the "ops" host, in which case, I might be inclined to simply name it `DEFAULT_EARTHDATA_HOST`.

What do you prefer?